### ISSUE TITLE
Add ability for IATI Admin users to delete an organisation from IATI Publisher

### ISSUE THREAD

**Issue by emmajclegg** [younginnovations/iatipublisher#1384]
Can "IATI Admin" users (i.e. super-admins) be given the ability to delete an organisation and its data from IATI Publisher? e.g. from the [Organisation List](https://publisher.iatistandard.org/list-organisations) screen.

![Image](https://github.com/younginnovations/iatipublisher/assets/137316374/e1df2ecf-238c-4055-a7b9-bea399deef3d)

This will help us:
- tackle an increase in spam accounts that are being created (we need to be able to easily delete these)
- help publishers who have made mistakes and want to start over re-registering their organisation in Publisher
- delete test organisation accounts that IATI Admins have created on the staging site which are no longer required

To try to prevent more spam accounts being created, we would also like to block certain email addresses from being able to create an account in Publisher (I've raised a separate GitHub issue for this: #1385 )

**Comment by PG-Momik** [comment]
@emmajclegg This seems doable. We could add a simple delete button next to `proxy` button. Clicking on the delete button would trigger a confirmation modal and IATI Admin can proceed accordingly.

Initial concerns regarding changes for this are:
1. Should there be an option to restore deleted organization and related data ?
2. Providing an option to restore an organization would mean we'd only be soft deleting said organization. 
3. Implementing soft delete and providing an option to restore would mean that  we'd also need to work on a different listing page , namely: `Restore organization listing page`
4. Not implementing soft delete would mean that once deleted, the data can never be restored.
5. With changes requested on #1385, should there be an option to mark emails as spam for user  associated to the organization  (that will be deleted).

I'll update this issue with possible tasks.

**Comment by emmajclegg** [comment]
@PG-Momik - a hard delete of the organisation would be fine (i.e. we don't need the option to restore data). I mainly see this being used to delete spam accounts. 

This did raise the question of backups - how often and where is Publisher data backed up? 

> We could add a simple delete button next to proxy button. Clicking on the delete button would trigger a confirmation modal and IATI Admin can proceed accordingly.

Sounds great. 

> With changes requested on https://github.com/younginnovations/iatipublisher/issues/1385, should there be an option to mark emails as spam for user associated to the organization (that will be deleted).

Yes, this sounds sensible, and it would avoid the need for admins to go to the User page separately to block the associated email address.

**Comment by PG-Momik** [comment]
@emmajclegg we're using digital ocean to host our database. Digital ocean allows us to restore database to any point in time of last 7 days.

**Comment by sarinasindurakar** [comment]
@PG-Momik 

- [ ] 1. Activity dataset and Organisation dataset  is not removed form IATI registry after deleting the organisation.
- [ ] 2. The users of the deleted organisation are not deleted from database .

**Comment by PG-Momik** [comment]
@emmajclegg 

This has been deployed to staging.

**Comment by emmajclegg** [comment]
@PG-Momik - I've tested deleting my test accounts then re-registering one on Publisher. Dataset deletion and the email block looks to work well. A few observations (these may or may not be related to this issue):

1) When I tried to delete my organisation "emma-troubleshooting" (with publisher ID _emmac-test_) from Publisher staging I get an error message:



![Image](https://github.com/younginnovations/iatipublisher/assets/137316374/b2d08603-6f75-4d0c-b397-143d6614b45e)


![Image](https://github.com/younginnovations/iatipublisher/assets/137316374/3597a0ba-887b-4d79-b6d2-23e183d58c29)


2) When I deleted, then re-registered another test organisation ("Emma's test publisher", with publisher ID _emma-test-publisher_), I could recreate an IATI Publisher account ok...

![Image](https://github.com/younginnovations/iatipublisher/assets/137316374/db196e38-acde-49a5-8e54-3717f6c6b6c9)



![Image](https://github.com/younginnovations/iatipublisher/assets/137316374/141bc53c-5ac0-4aa9-a919-cd79623fc4c6)



....BUT my publisher ID in Settings is now blank and not editable. 

![Image](https://github.com/younginnovations/iatipublisher/assets/137316374/f0e1a1aa-f48e-4b2d-8921-6f6865d05b6f)


I know we won't necessarily expect organisations to reregister, but I wondered if the blank publisher ID is related to changes by #1298

**Comment by PG-Momik** [comment]
@emmajclegg 

Most of the issues listed by you stem from #1298

**Comment by PG-Momik** [comment]
@emmajclegg 

 After your [comment on 1385](https://github.com/younginnovations/iatipublisher/issues/1385#issuecomment-2008357042) , marking users as spam via user listing had not been worked on. The 'Mark as spam' option on the users listing page was experimental code that had been removed from this issue's code, but it seems staging had some residue code. I've redeployed staging.


![Image](https://github.com/younginnovations/iatipublisher/assets/57512059/079112d0-afcb-4bb8-a80b-cf54d5ce4a0c)
